packages/device_info/lib/device_info.dart

```diff
@@ -0,0 +1,231 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
```

Comment:
Thanks, @mehmetf for the feedback. I don't see how the Firebase or Crashlytics API would let me encode and send a Dart stacktrace. Until we figure that out, I'll stick to recommending Sentry.io, which supports custom languages, stacktraces, fingerprints, etc. For native stacktraces and hard crashes, definitely an out-of-the-box solution is the way to go. Here's an example report I submitted using the [Sentry.io client](https://pub.dartlang.org/packages/sentry) I built recently:

Dart stacktrace encoded in a way Sentry.io can understand and format:

<img width="979" alt="screen shot 2017-06-29 at 10 52 34 am" src="https://user-images.githubusercontent.com/211513/27706557-d2aed236-5cc6-11e7-9b1c-60a926f8eef0.png">

Crash report grouping based on stable fingerprint:

<img width="1341" alt="screen shot 2017-06-29 at 10 52 13 am" src="https://user-images.githubusercontent.com/211513/27706561-d7ac9e30-5cc6-11e7-9455-02cf2ab01666.png">